nano/lib/blocks.hpp

```diff
@@ -63,6 +106,9 @@ class block
 
 protected:
 	mutable nano::block_hash cached_hash{ 0 };
+	// Contextual details about a block, some fields may or may not be set depending on block type
+	// This field is set via sideband_set in ledger processing or deserializing blocks from the database
+	boost::optional<nano::block_sideband> sideband_m{ boost::none };
```

Comment:
As optional is a value-wrapper, it looks like this will add 100 bytes to every block (block_sideband+block_details), also when sideband is not present (actually +8 bytes extra overhead for optional on my toolchain) 
I wonder if it's better to use a unique_ptr here (which has optional semantics, sort of), possibly using polymorphic sideband types to potentially save more space (may not be worth the inconvenience on call sites, esp since presumably most blocks will be state)

Reply:
`unique_ptr` sounds good, I had a polymorphic sideband as a future task 👍 if you agree with doing it later, as this PR is quite large already.

Reply:
Now using `nano::optional_ptr`, thanks for adding it!